src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/bqsr/BaseRecalibratorFn.java

```diff
@@ -1,79 +1,57 @@
 import org.broadinstitute.hellbender.tools.recalibration.covariates.StandardCovariateList;
 import org.broadinstitute.hellbender.tools.walkers.bqsr.ReadRecalibrationInfo;
 import org.broadinstitute.hellbender.tools.walkers.bqsr.RecalibrationEngine;
 import org.broadinstitute.hellbender.transformers.ReadTransformer;
 import org.broadinstitute.hellbender.utils.BaseUtils;
 import org.broadinstitute.hellbender.utils.MathUtils;
-import org.broadinstitute.hellbender.utils.Utils;
 import org.broadinstitute.hellbender.utils.baq.BAQ;
 import org.broadinstitute.hellbender.utils.clipping.ReadClipper;
 import org.broadinstitute.hellbender.utils.read.AlignmentUtils;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
 import org.broadinstitute.hellbender.utils.read.ReadUtils;
 import org.broadinstitute.hellbender.utils.recalibration.EventType;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.variant.Variant;
 
 import java.io.File;
 import java.io.FileNotFoundException;
+import java.io.IOException;
 import java.io.PrintStream;
 import java.util.Arrays;
-import java.util.List;
 
 /**
- * First pass of the base quality score recalibration -- Generates recalibration table based on various covariates
- * (such as read group, reported quality score, machine cycle, and nucleotide context).
- *
- * Based off of BaseRecalibrator, but changed to be able to work in Dataflow.
- *
- * Expected usage:
- * -ship the command line to Dataflow workers
- * -create a BaseRecalibratorWorker there,
- * -feed it the right inputs (use the CalibrationTablesBuilder wrapper
- *  to help)
- * -extract the outputs, merge them, and save.
- * -use a BaseRecalibratorWorker to format the outputs into a report.
+ * DoFn for BaseRecalibrator. Takes in reads, spits out RecalibrationTables.
  */
+public final class BaseRecalibratorFn extends DoFnWLog<KV<GATKRead, ReadContextData>, RecalibrationTables> {
+    private static final long serialVersionUID = 1L;
 
-@CommandLineProgramProperties(
-        summary = "First pass of the Base Quality Score Recalibration (BQSR) -- Generates recalibration table based on various user-specified covariates (such as read group, reported quality score, machine cycle, and nucleotide context).",
-        oneLineSummary = "Generates recalibration table (do not run this one from the cmd line)",
-        programGroup = ReadProgramGroup.class
-)
-public final class BaseRecalibratorWorker {
-    protected static final Logger logger = LogManager.getLogger(BaseRecalibratorWorker.class);
+    private PCollectionView<SAMFileHeader> headerView;
+    private transient SAMFileHeader header;
+    private PCollectionView<SAMSequenceDictionary> referenceSequenceDictionaryView;
+    private transient SAMSequenceDictionary referenceSequenceDictionary;
 
     @ArgumentCollection(doc="all the command line arguments for BQSR and its covariates")
     private BaseRecalibrationArgumentCollection BRAC;
```

Comment:
Does not need to be annotated as an `@ArgumentCollection`, since this is no longer tied to the command line parsing (it's just storage).